### PR TITLE
v0.4 integration with SSO

### PR DESCRIPTION
**Pull request by simonstrong-dbt**
## Context

Upgrading Sandbox env to latest release of redbox.

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: django_app/redbox_app/settings.py [hunk 1]
```diff
@@ -144,8 +153,8 @@
 SITE_ID = 1
 AUTH_USER_MODEL = "redbox_core.User"
 ACCOUNT_EMAIL_VERIFICATION = "none"
-LOGIN_REDIRECT_URL = "homepage"
-LOGIN_URL = "sign-in"
+# LOGIN_REDIRECT_URL = "homepage"
```
Comment (a-gleeson) [inline django_app/redbox_app/settings.py:156]: can remove comment?

File: django_app/redbox_app/redbox_core/auth_views.py [hunk 2]
```diff
@@ -15,6 +16,8 @@
 def sign_in_view(request: HttpRequest):
     if request.user.is_authenticated:
         return redirect("homepage")
+    if settings.LOGIN_METHOD == "sso":
```
Comment (a-gleeson) [inline django_app/redbox_app/redbox_core/auth_views.py:19]: should we log the log in method?

File: redbox-core/redbox/models/settings.py
Comment (isobel-daley-6point6) [inline redbox-core/redbox/models/settings.py:1]: Are any changes required at line 188 to ensure that elastic local settings are used? @simonstrong-dbt

**Comment by a-gleeson** [comment]
I have an idea to make this more configurable we should leave the `User` model as the other model but create our own `SSOUser` or something. Then in the settings configure line 145 `AUTH_USER_MODEL = "redbox_core.User"` this will allow us to switch the db model for the user between services. We should make sure when we reference the user's User model it should be via `get_user_model` https://docs.djangoproject.com/en/5.0/topics/auth/customizing/#referencing-the-user-model

**Comment by simonstrong-dbt** [comment]
> I have an idea to make this more configurable we should leave the `User` model as the other model but create our own `SSOUser` or something. Then in the settings configure line 145 `AUTH_USER_MODEL = "redbox_core.User"` this will allow us to switch the db model for the user between services. We should make sure when we reference the user's User model it should be via `get_user_model` https://docs.djangoproject.com/en/5.0/topics/auth/customizing/#referencing-the-user-model

@a-gleeson I initially looked at doing this so I didn't affect the original model, but there was a reason I moved away from it again.  I'd have to double check as it was a while ago.  If we use SSOUser as our class, when an object of that is created, even with AUTH_USER_MODEL set to it, won't other functions that call on that object not be able to find it? e.g. Line 196 in models.py (in django_app) `user = models.ForeignKey(User, on_delete=models.CASCADE)` references the User class.  If that's not present (because it's SSOUser) this won't work?

**Comment by simonstrong-dbt** [comment]
@a-gleeson  sorry didn't read the link - also should have thought about something like that before.. less haste more speed?

**Comment by a-gleeson** [comment]
It's an easy miss, I only know from multiple Django projects recently!

**Pull request merged**